src/app/admin/subjects/subjects.component.ts

```diff
@@ -39,10 +55,52 @@ export class SubjectsComponent implements OnInit {
           return of(null);
         })
       )
-      .subscribe((newData: SSubjects | null) => {
+      .subscribe((newData: Array<ISubjects> | null) => {
+        if (newData) {
+          this.subjectTableList = [...this.subjectTableList, ...newData];
+          this.dataSource.sort = this.sort;
+        }
+      });
+  }
+  edit(row: ISubjects): void{
+    const newDialogSubject = this.dialog.open(SubjectsCreateModalComponent, {
+      width: '500px',
+      disableClose: true,
+    });
+    newDialogSubject.afterClosed()
+      .pipe(
+        mergeMap((data) => {
+          if (data) {
+            return this.subjectsService.updateSubject(row.subject_id, data);
+          }
+          return of(null);
+        })
+      )
+      .subscribe((newData: Array<ISubjects> | null) => {
         if (newData) {
-          this.subjectTableList = [newData, ...this.subjectTableList];
+          this.showSubjects();
+          this.dataSource.sort = this.sort;
         }
       });
   }
+
+  delete(row: ISubjects): void {
+    const dialogData = `Ви видаляєте ${row.subject_name}, ви впевнені?`;
+    const dialogRef = this.dialog.open(SubjectConfirmComponent, {
+      maxWidth: '400px',
+      data: dialogData
+    });
+    dialogRef.afterClosed().subscribe(dialogResult => {
+      this.result = dialogResult;
+      if (this.result) {
```

Comment:
why are you storing this.result?, why not simply use if (dialogResult) {}

Reply:
you can do it in a fancier way 
`dialogRef.afterClosed(
 filter(res => res), 
 switchMap(() => this.subjectService.deleteSubject(row.subject_id))).subscribe((res) => this.subjectTableList = this.subjectTableList.filter(item => item.subject_id !== row.subject_id);)`

but it's up to you